physics/module_mp_thompson.F90

```diff
@@ -433,554 +430,554 @@ MODULE module_mp_thompson
 
 !>  - Compute snow collecting cloud ice
-      t1_qs_qi = PI*.25*av_s
+         t1_qs_qi = PI*.25*av_s
 
 !>  - Compute evaporation of rain; ignore depositional growth of rain
-      t1_qr_ev = 0.78 * crg(10)
-      t2_qr_ev = 0.308*Sc3*SQRT(av_r) * crg(11)
+         t1_qr_ev = 0.78 * crg(10)
+         t2_qr_ev = 0.308*Sc3*SQRT(av_r) * crg(11)
 
 !>  - Compute sublimation/depositional growth of snow
-      t1_qs_sd = 0.86
-      t2_qs_sd = 0.28*Sc3*SQRT(av_s)
+         t1_qs_sd = 0.86
+         t2_qs_sd = 0.28*Sc3*SQRT(av_s)
 
 !>  - Compute melting of snow
-      t1_qs_me = PI*4.*C_sqrd*olfus * 0.86
-      t2_qs_me = PI*4.*C_sqrd*olfus * 0.28*Sc3*SQRT(av_s)
+         t1_qs_me = PI*4.*C_sqrd*olfus * 0.86
+         t2_qs_me = PI*4.*C_sqrd*olfus * 0.28*Sc3*SQRT(av_s)
 
 !>  - Compute sublimation/depositional growth of graupel
-      t1_qg_sd = 0.86 * cgg(10)
-      t2_qg_sd = 0.28*Sc3*SQRT(av_g) * cgg(11)
+         t1_qg_sd = 0.86 * cgg(10)
+         t2_qg_sd = 0.28*Sc3*SQRT(av_g) * cgg(11)
 
 !>  - Compute melting of graupel
-      t1_qg_me = PI*4.*C_cube*olfus * 0.86 * cgg(10)
-      t2_qg_me = PI*4.*C_cube*olfus * 0.28*Sc3*SQRT(av_g) * cgg(11)
+         t1_qg_me = PI*4.*C_cube*olfus * 0.86 * cgg(10)
+         t2_qg_me = PI*4.*C_cube*olfus * 0.28*Sc3*SQRT(av_g) * cgg(11)
 
 !>  - Compute constants for helping find lookup table indexes
-      nic2 = NINT(ALOG10(r_c(1)))
-      nii2 = NINT(ALOG10(r_i(1)))
-      nii3 = NINT(ALOG10(Nt_i(1)))
-      nir2 = NINT(ALOG10(r_r(1)))
-      nir3 = NINT(ALOG10(N0r_exp(1)))
-      nis2 = NINT(ALOG10(r_s(1)))
-      nig2 = NINT(ALOG10(r_g(1)))
-      nig3 = NINT(ALOG10(N0g_exp(1)))
-      niIN2 = NINT(ALOG10(Nt_IN(1)))
+         nic2 = NINT(ALOG10(r_c(1)))
+         nii2 = NINT(ALOG10(r_i(1)))
+         nii3 = NINT(ALOG10(Nt_i(1)))
+         nir2 = NINT(ALOG10(r_r(1)))
+         nir3 = NINT(ALOG10(N0r_exp(1)))
+         nis2 = NINT(ALOG10(r_s(1)))
+         nig2 = NINT(ALOG10(r_g(1)))
+         nig3 = NINT(ALOG10(N0g_exp(1)))
+         niIN2 = NINT(ALOG10(Nt_IN(1)))
 
 !>  - Create bins of cloud water (from min diameter up to 100 microns)
-      Dc(1) = D0c*1.0d0
-      dtc(1) = D0c*1.0d0
-      do n = 2, nbc
-         Dc(n) = Dc(n-1) + 1.0D-6
-         dtc(n) = (Dc(n) - Dc(n-1))
-      enddo
+         Dc(1) = D0c*1.0d0
+         dtc(1) = D0c*1.0d0
+         do n = 2, nbc
+            Dc(n) = Dc(n-1) + 1.0D-6
+            dtc(n) = (Dc(n) - Dc(n-1))
+         enddo
 
 !>  - Create bins of cloud ice (from min diameter up to 2x min snow size)
-      xDx(1) = D0i*1.0d0
-      xDx(nbi+1) = 2.0d0*D0s
-      do n = 2, nbi
-         xDx(n) = DEXP(DFLOAT(n-1)/DFLOAT(nbi) &
-                  *DLOG(xDx(nbi+1)/xDx(1)) +DLOG(xDx(1)))
-      enddo
-      do n = 1, nbi
-         Di(n) = DSQRT(xDx(n)*xDx(n+1))
-         dti(n) = xDx(n+1) - xDx(n)
-      enddo
+         xDx(1) = D0i*1.0d0
+         xDx(nbi+1) = 2.0d0*D0s
+         do n = 2, nbi
+            xDx(n) = DEXP(DFLOAT(n-1)/DFLOAT(nbi) &
```

Comment:
DEXP, DLOG, DSQRT are archaic. EXP, LOG and SQRT return values of the type you provide, so you could use the standard intrinsics directly by passing inputs of type kind_dbl_prec. 
You can either define the locals are double, or you could convert them as you pass them to the intrinsic. So DLOG(var) -> LOG(real(var,kind=kind_dbl_phys))